core-plugins/src/main/java/io/cdap/plugin/batch/joiner/JoinerConfig.java

```diff
@@ -163,8 +203,55 @@ public Schema getOutputSchema(FailureCollector collector) {
   }
 
   boolean requiredPropertiesContainMacros() {
-    return containsMacro(SELECTED_FIELDS) || containsMacro(REQUIRED_INPUTS) || containsMacro(JOIN_KEYS) ||
-      containsMacro(OUTPUT_SCHEMA);
+    return containsMacro(SELECTED_FIELDS) || containsMacro(REQUIRED_INPUTS) ||
+      containsMacro(OUTPUT_SCHEMA) || containsMacro(CONDITION_TYPE) ||
+      (BASIC.equalsIgnoreCase(conditionType) && containsMacro(JOIN_KEYS)) ||
+      (ADVANCED.equalsIgnoreCase(conditionType) && containsMacro(CONDITION_EXPR));
+  }
+
+  private JoinCondition.Op getConditionType(FailureCollector failureCollector) {
+    if (conditionType == null || conditionType.isEmpty() || BASIC.equals(conditionType)) {
+      return JoinCondition.Op.KEY_EQUALITY;
+    }
+    if (ADVANCED.equalsIgnoreCase(conditionType)) {
+      return JoinCondition.Op.EXPRESSION;
+    }
+    failureCollector.addFailure("Invalid condition type " + conditionType, "Set it to basic or advanced.");
```

Comment:
nit: add quotes around `basic` and `advanced` to make it clear that these are the expected values.